### DISCUSSION TITLE
maxGraph source code layout diagrams

### DISCUSSION THREAD

**Discussion by mcyph** [maxGraph/maxGraph#72] [Development]
I've started to make some basic diagrams which I hope can be a starting point for documentation. Ideally, I'd like for these to be written in simple English with as little domain-specific vocabulary as possible to make it easier for new users to get started.

I'm still not sure what some of the classes do myself, but I've tried to explain things as best as I understood them. Please let me know if you think things could be better explained or if corrections need to be made. If anyone has time, it would be great if you could do this for one or more of the other folders as in the `development` branch and post a response. 

![maxgraph drawio](https://user-images.githubusercontent.com/20507948/149648211-b0ee5aca-9e80-4248-95d2-6d3dd9e35e57.png)

![maxgraph_handlers drawio](https://user-images.githubusercontent.com/20507948/149648220-12f2c14e-3679-450a-b8e5-6fde8312baed.png)

**Comment by tbouffard** [comment]
Nice, I love this idea.
@mcyph can you tell me how you create the diagram? is it pure svg or did you use a specific tool? If you can your work in a branch this would be perfect.

I guess it will be easier for maintenance to use the same way.

**Reply by mcyph** [reply]
It's just the plain stencils from https://app.diagrams.net/, I figured I may as well use it because of it using mxGraph, but even dotpoints would be a good start. I'm sure they could be better laid out than how they are currently so feel free to move things around, will push to dev!
